clang/lib/CodeGen/CodeGenModule.h

```diff
@@ -1586,6 +1586,12 @@ class CodeGenModule : public CodeGenTypeCache {
   void AddGlobalDtor(llvm::Function *Dtor, int Priority = 65535,
                      bool IsDtorAttrFunc = false);
 
+  // Return whether structured convergence intrinsics should be generated for
+  // this target.
+  bool shouldEmitConvergenceTokens() const {
+    return getTriple().isSPIRVLogical();
```

Comment:
Should add a TODO this should just be unconditional in the future

Reply:
Not necessarily. The SPIR-V compute (as OpenCL not Vulkan compute) doesn't require a structured CFG, so this could remain forever if they don't see any benefit in generating a SCFG.

Reply:
This doesn't have anything to do with what the target wants to do with with the CFG lowering. Structurally the IR should not allow uncontrolled convergence, and that it exists is a wart that needs to exist until everywhere handles convergence tokens

Reply:
That makes sense. Added a TODO line!